code/game/machinery/_machinery.dm

```diff
@@ -677,8 +677,17 @@
 	..()
 	RefreshParts()
 
-/obj/machinery/proc/RefreshParts() //Placeholder proc for machines that are built using frames.
-	return
+/obj/machinery/proc/RefreshParts()
+	SHOULD_CALL_PARENT(TRUE)
+	if(!component_parts || !component_parts.len)
+		return
+	var/parts_energy_rating = 0
+	for(var/obj/item/stock_parts/part in component_parts)
+		parts_energy_rating += part.energy_rating
+
+	idle_power_usage = initial(idle_power_usage) * (1 + parts_energy_rating)
+	active_power_usage = initial(active_power_usage) * (1 + parts_energy_rating)
+	update_current_power_usage()
```

Comment:
Shouldn't no parts still change power usage?

Reply:
Well how? If you have no parts you'd still have the initial values

Reply:
if I have parts, then lose some, I'll have my old numbers right?

Reply:
Ah but currently you cannot lose parts. Unless you want me to add a check for future safeguard there isn't a way to get lower level parts

Reply:
Or no part without deconstruction

Reply:
I'd like a safeguard.
The check also doesn't really make sense honestly, it confuses me

Reply:
i'll add the safeguard
what's wrong with the check?